infrastructure/storage/src/lmdb_store/store.rs

```diff
@@ -137,6 +137,22 @@ impl LMDBBuilder {
             target: LOG_TARGET,
             "({}) LMDB environment created with a capacity of {} MB.", path, self.db_size_mb
         );
+
+        // TODO: `env.set_mapsize(new_size * 1024 * 1024)` if `usage` approach the `self.db_size_mb` limit
+        let env_info = env.info()?;
+        let env_stat = env.stat()?;
+        let size_used = env_stat.psize as usize * env_info.last_pgno;
+        let space_remaining = env_info.mapsize - size_used;
+        let usage = (size_used as f64 / env_info.mapsize as f64) * 100.0;
+        debug!(
+            target: LOG_TARGET,
+            "({}) LMDB environment has {} MB remaining space with usage factor {:.*} %.",
+            path,
+            space_remaining / (1024 * 1024),
+            2,
+            usage
+        );
+
```

Comment:
If we can call set_mapsize at any time, why don't we just go ahead and call `set_mapsize` right here if usage factor > 0.8?

Reply:
Increase it by 1GB at a time?

Reply:
This check is only done at start up, so another PR might look at this periodically while the node is running

Reply:
Starting/syncing the blockchain from scratch will be safe. 
I would suggest `db_size_mb =  1 GB` for the `rincewind` db, but a restart using a set value might be problematic as it could open a 50 GB db with 2 GB worth of content and call `set_mapsize` with a smaller value of say 1 GB. `set_mapsize` works equally well reducing or increasing the db environment size. The code change I have in mind is:
- Find the db on disk if it exists, determine its actual size
- If db does not exist, use `db_size_mb =  1 GB` in call to `set_mapsize` 
- If  db exist, use biggest of `db_size_mb =  1 GB` and physical size in call to `set_mapsize` 
- Determine the usage, and adjust size if required
- For startup I would recommend this as a function of the pre-set size of say 1 GB. If we have less than 50% of the pre-set size left, increase by 100% of the of the pre-set size.

Reply:
Running a test - seems LMDB is forgiving. It does not crash if `set_mapsize` is smaller than the actual data content, `env.info().mapsize` only shrinks to the actual data size.

Reply:
So the  code changes seems to work fine in Windows, will still test in Linux:
```
2020-05-06 10:33:15.262362700 [lmdb] DEBUG (rincewind\db) LMDB environment usage factor 99.99 %., size used 56 MB, increased by 40 MB.
2020-05-06 10:33:15.262362700 [lmdb] INFO  (rincewind\db) LMDB environment created with a capacity of 96 MB, 40 MB remaining.
```

Reply:
Please see updated code changes